src/metadata/metadata.c

```diff
@@ -572,6 +571,7 @@ static int ocf_metadata_init_fixed_size(struct ocf_cache *cache,
 		cache->user_parts[i].runtime = &part_runtime[i];
 		cache->user_parts[i].id = i;
 	}
+	cache->free = &part_runtime[PARTITION_INVALID];
```

Comment:
This looks terrible. We should have more meaningful define for that. And at one go get rid of this "+2" few lines above.

Reply:
I would appreciate more constructive comments. INVALID makes some sense since the cachelines in this partition are invalid. However after some more thinking I realized that inserted cachelines can also be invalid, so FREE would probably be more precise.